include/s3select.h

```diff
@@ -529,10 +535,12 @@ struct s3select : public bsc::grammar<s3select>
       alias_name = bsc::lexeme_d[(+bsc::alpha_p >> *bsc::digit_p)] ;
 
 
-      when_case_else_projection = (bsc::str_p("case")  >> (+when_stmt) >> bsc::str_p("else") >> arithmetic_expression >> bsc::str_p("end")) [BOOST_BIND_ACTION(push_case_when_else)];
+      when_case_else_projection = (bsc::str_p("case")  >> (+when_stmt) | (+when_stmt_simple) >> bsc::str_p("else") >> arithmetic_expression >> bsc::str_p("end")) [BOOST_BIND_ACTION(push_case_when_else)];
 
       when_stmt = (bsc::str_p("when") >> condition_expression >> bsc::str_p("than") >> arithmetic_expression)[BOOST_BIND_ACTION(push_when_than)];
 
+      when_stmt_simple = (projections >> bsc::str_p("when") >> arithmetic_expression >> bsc::str_p("then") >> arithmetic_expression)[BOOST_BIND_ACTION(push_when_then_simple)];
```

Comment:
not correct.
/projections/ is not correct for case-when , it accept also other syntax not to fit to our case.
should use /arithmetic_expression/